server/src/test/java/org/elasticsearch/cluster/coordination/AtomicRegisterCoordinatorTests.java

```diff
@@ -682,5 +641,53 @@ long getInitialTermBeforeJoiningALeader() {
         public void close() {
             assertTrue(openPersistedStates.remove(this));
         }
+
+        @Override
+        public void getLatestStoredState(ActionListener<ClusterState> listener) {
+            ActionListener.completeWith(listener, () -> {
+                var latestClusterState = sharedStore.getLatestClusterState();
+                if (latestClusterState == null) {
+                    return null;
+                }
+
+                if (isLatestAcceptedStateStale(latestClusterState) == false) {
+                    return null;
+                }
+
+                if (latestClusterState.term() > currentTerm) {
+                    return null;
+                }
+
+                return ClusterStateUpdaters.recoverClusterBlocks(
+                    ClusterStateUpdaters.addStateNotRecoveredBlock(
+                        ClusterState.builder(new ClusterName("elasticsearch"))
+                            .metadata(
+                                Metadata.builder(latestClusterState.state())
+                                    .coordinationMetadata(
+                                        new CoordinationMetadata(
+                                            latestClusterState.term(),
+                                            // Keep the previous configuration so the assertions don't complain about
+                                            // a different configuration, we'll change it right away
+                                            latestAcceptedState.getLastAcceptedConfiguration(),
+                                            latestAcceptedState.getLastCommittedConfiguration(),
+                                            Set.of()
+                                        )
+                                    )
+                            )
+                            .version(latestClusterState.version())
+                            .blocks(latestAcceptedState.blocks())
```

Comment:
We shouldn't need this now that we're loading the blocks from the metadata right?

```suggestion
```